script.js

```diff
@@ -1,4 +1,220 @@
+    return `Sorry! ${userName}, You Lost! You chose ${input}, secret word is ${computerChoice}<br><br>${defaultClosingMessage}`;
+  }
+  return myOutputValue;
+};
+
+// refactor korean SPS
+var playKoreanSPS = function (userName, input) {
+  var myOutputValue = ``;
+  var computerChoice = sPS();
+  // var winnerOfThisRound = ``;
+  // validation to inform user can only input SPS
+  if (!(input == SCISSORS || input == PAPER || input == STONE))
+    return `Pls enter scissors, paper or stone!`;
+
+  // defining win conditions for Korean SPS
+  if (
+    (input == SCISSORS && computerChoice == PAPER) ||
+    (input == PAPER && computerChoice == STONE) ||
+    (input == STONE && computerChoice == SCISSORS)
+  ) {
+    winnerOfThisRound = userName;
+    return `K<br>${defaultIntroMessage}K <br>Congrats ${userName}, you won this turn!<br> <br> Your choice: ${input} <br> vs <br> Computer choice: ${computerChoice} <br><br><br>Cheers! <br>Most recent winner is ${winnerOfThisRound}<br> ${defaultClosingMessage}`;
+  }
+
+  // defining lose conditions for Korean SPS
+  if (
+    (input == SCISSORS && computerChoice == STONE) ||
+    (input == PAPER && computerChoice == SCISSORS) ||
+    (input == STONE && computerChoice == PAPER)
+  ) {
+    winnerOfThisRound = `computer`;
+    return `K<br>${defaultIntroMessage}K<br> Oops ${userName}, you lost this turn! <br> <br> Your choice: ${input} <br> vs <br> Computer choice: ${computerChoice}<br><br><br> Better luck next time!<br> Most recent winner is ${winnerOfThisRound}<br> ${defaultClosingMessage}
+    `;
+  }
+  mostRecentWinner = winnerOfThisRound;
+  // defining draw condition for Korean SPS
+  if (input == computerChoice) {
+    return `K<br>${defaultIntroMessage}K<br>It's a DRAW, ${userName}! <br><br> Your choice: ${input} <br> vs <br> Computer choice: ${computerChoice} <br><br><br>Pls try again <br> Ultimate winner of the round is +++${mostRecentWinner}+++<br> ${defaultClosingMessage}`;
+  }
+
+  return myOutputValue;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = ``;
+
+  // set initial stage for user to input name
+  if (currentGameMode == `waiting for user name`) {
+    // set user name
+    userName = input;
+    currentGameMode = `SPS`;
+    return `Hello ${userName}! Good luck! <br><br>Input scissors, paper or stone to start normal SPS. <br><br>Alternatively input any of the following to change game mode: korean, reverse, secret`;
+  }
+  if (currentGameMode == `SPS`) {
+    myOutputValue = playSPS(userName, input);
+  }
+
+  if (input == `normal`) {
+    totalTries = 0;
+    winRecord = 0;
+    drawRecord = 0;
+    currentGameMode = `SPS`;
+    return `Welome to normal mode! Input scissors, paper or stone to start`;
+  }
+
+  if (input == `reverse`) {
+    totalTries = 0;
+    winRecord = 0;
+    drawRecord = 0;
+    currentGameMode = `reversed SPS`;
+    return `Welome to reversed mode! Input scissors, paper or stone to start`;
+  }
```

Comment:
To organise better, all the ifs for game mode changes can be put at the top
for playing the game if (currentGameMode==... ), these can be put after